engine/src/witness/arb.rs

```diff
@@ -0,0 +1,175 @@
+		+ Sync
+		+ Clone
+		+ 'static,
+	ProcessingFut: Future<Output = ()> + Send + 'static,
+{
+	let key_manager_address = state_chain_client
+		.storage_value::<pallet_cf_environment::ArbitrumKeyManagerAddress<state_chain_runtime::Runtime>>(
+			state_chain_client.latest_finalized_block().hash,
+		)
+		.await
+		.context("Failed to get KeyManager address from SC")?;
+
+	let vault_address = state_chain_client
+		.storage_value::<pallet_cf_environment::ArbitrumVaultAddress<state_chain_runtime::Runtime>>(
+			state_chain_client.latest_finalized_block().hash,
+		)
+		.await
+		.context("Failed to get Vault contract address from SC")?;
+
+	let address_checker_address = state_chain_client
+		.storage_value::<pallet_cf_environment::ArbitrumAddressCheckerAddress<state_chain_runtime::Runtime>>(
+			state_chain_client.latest_finalized_block().hash,
+		)
+		.await
+		.expect(STATE_CHAIN_CONNECTION);
+
+	let supported_arb_erc20_assets: HashMap<cf_primitives::chains::assets::arb::Asset, H160> =
+		state_chain_client
+			.storage_map::<pallet_cf_environment::ArbitrumSupportedAssets<state_chain_runtime::Runtime>, _>(
+				state_chain_client.latest_finalized_block().hash,
+			)
+			.await
+			.context("Failed to fetch Arbitrum supported assets")?;
+
+	let usdc_contract_address = *supported_arb_erc20_assets
+		.get(&cf_primitives::chains::assets::arb::Asset::ArbUsdc)
+		.context("ArbitrumSupportedAssets does not include USDC")?;
+
+	let supported_arb_erc20_assets: HashMap<H160, cf_primitives::Asset> =
+		supported_arb_erc20_assets
+			.into_iter()
+			.map(|(asset, address)| (address, asset.into()))
+			.collect();
+
+	let arb_source = EvmSource::<_, Arbitrum>::new(arb_client.clone())
+		.strictly_monotonic()
+		.shared(scope);
+
+	arb_source
+		.clone()
+		.chunk_by_time(epoch_source.clone(), scope)
+		.chain_tracking(state_chain_client.clone(), arb_client.clone())
+		.logging("chain tracking")
+		.spawn(scope);
+
+	let vaults = epoch_source.vaults::<Arbitrum>().await;
+
+	// ===== Full witnessing stream =====
+
+	let arb_safety_margin = state_chain_client
+		.storage_value::<pallet_cf_ingress_egress::WitnessSafetyMargin<
+			state_chain_runtime::Runtime,
+			state_chain_runtime::ArbitrumInstance,
+		>>(state_chain_stream.cache().hash)
+		.await?
+		// Default to berghain in case the value is missing (e.g. during initial upgrade)
+		.unwrap_or(ARBITRUM_SAFETY_MARGIN);
+
+	tracing::info!("Safety margin for Arbitrum is set to {arb_safety_margin} blocks.",);
+
+	let arb_safe_vault_source = arb_source
+		.lag_safety(arb_safety_margin as usize)
+		.logging("safe block produced")
+		.chunk_by_vault(vaults, scope);
+
+	let arb_safe_vault_source_deposit_addresses = arb_safe_vault_source
+		.clone()
+		.deposit_addresses(scope, state_chain_stream.clone(), state_chain_client.clone())
+		.await;
+
+	arb_safe_vault_source
+		.clone()
+		.key_manager_witnessing(process_call.clone(), arb_client.clone(), key_manager_address)
+		.continuous("KeyManager".to_string(), db.clone())
```

Comment:
So we need to have unique names, so maybe we add a parameter to continous to add in the Chain (as previous) i.e. format!("{}-{}", ...), but only use if for Arb at teh moment. @kylezs

Reply:
So old Eth DB can be used

Reply:
I think I'd rather just use something like "Arbitrum KeyManager" manually here - and not have the name change in continuous, then create an issue to do a proper migration where they're all namespaced by chain.

Reply:
Yeah happy with that, if this is done, happy for this to be merged.

Reply:
I have prefixed the names passed in ```continuous``` and ```logging``` functions by Arbitrum and Ethereum (for etherreum)